File: C_lang_Foundation_of_Programming/30_Second_Min_from_N.c
//write a program which accept N Numbers from user and print second last minimum number from that numbers.
//INPUT: 4 7 2 1 5 3 8
//OUPUT: 2

#include<stdio.h>

int main()
{
	int iNO=0,i=0,imin=0,isecMin=0,itemp=0;

	printf("Program to Find Second Minimum.\n");

	printf("Enter Total Numbers(N):");
	scanf("%d",&iNO);

	printf("Enter Numbers :\n");
	
	scanf("%d",&itemp); //accept first input to initialise imin and isecMin

	imin=isecMin=itemp;

	for(i=1;i<iNO;i++)
	{
		scanf("%d",&itemp);

		if(imin > itemp) 
		{
			isecMin=imin;
			imin=itemp;

		}
		if(imin == isecMin)
		{
			isecMin=itemp;
		}
		if(itemp < isecMin && itemp > imin)
		{
			isecMin=itemp;
		}
	}
	if(imin == isecMin)  // if all numbers are smae then there is no second max
	{
		isecMin=-1;
	}
	printf("Second min :%d",isecMin);
	return 0;
}

File: Python_Foundation_of_Programming/30_second_min_from_n.py
#write a program which accept N Numbers from user and print second minimum number from that numbers.
#input:N=7 4 7 2 1 5 3 8 
#output:2

iN = int(input("Enter how many numbers?"))

lint_list= []

#accept input
for i in range(0,iN):
    itemp= int(input("Enter {}:".format(i+1)))
    lint_list.append(itemp)

#find second min                          <-- way 1
imin = isec_min =lint_list[0]

for i in range(0,iN):
    if(imin > lint_list[i]):
        isec_min= imin
        imin = lint_list[i]
    elif lint_list[i] > imin and lint_list[i] < isec_min:
        isec_min=lint_list[i]
    elif imin == isec_min:
        isec_min= lint_list[i]
    print("isec_max",isec_min)
print("Second min :",isec_min)

#find second min                    <-- way 2

lint_list.sort()
ltemp_list = lint_list
imin = ltemp_list[0]
isec_min=False

for i in range(0,iN):
    if imin != ltemp_list[i]:
        isec_min=ltemp_list[i]
        break
print("Second minimum:",isec_min)
